Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java

```diff
@@ -269,4 +271,87 @@ public boolean deleteCertificateFromTrustStore(String alias) throws
+                is = new FileInputStream(storeLoc);
+            }
+            tstore.load(is, passkey.toCharArray());
+
+            if (tstore.containsAlias(oldAlias)) {
+                certToRestore = cert;
+                os = updateCert(oldAlias, cert, storeLoc, passkey, tstore);
+                isUpdateSuccessful = true;
+            }
+        } catch (final IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException ex) {
+            isUpdateSuccessful = false;
+            LOG.error("Unable to update the Certifiate: ", ex.getLocalizedMessage(), ex);
+            throw new CertificateManagerException(ex.getMessage(), ex);
+        } finally {
+            closeFiles(is, os);
+        }
+        return isUpdateSuccessful;
+    }
+
+    @Override
+    public boolean updateCertificateKS(String oldAlias, Certificate cert) throws CertificateManagerException {
+
+        final Map<String, String> keyStoreProperties = cmHelper.getKeyStoreSystemProperties();
+        final String storeType = keyStoreProperties.get(KEY_STORE_TYPE_KEY);
+        final String storeLoc = keyStoreProperties.get(KEY_STORE_KEY);
+        final String passkey = keyStoreProperties.get(KEY_STORE_PASSWORD_KEY);
+
+        FileInputStream is = null;
+        FileOutputStream os = null;
+        boolean isUpdateSuccessful = false;
+        try {
+            KeyStore keyStore = cmHelper.getKeyStore();
+            if (!PKCS11_TYPE.equalsIgnoreCase(storeType)) {
+                is = new FileInputStream(storeLoc);
+            }
+            keyStore.load(is, passkey.toCharArray());
+            keyStore.containsAlias(cert.getAlias());
+            if (keyStore.containsAlias(oldAlias)) {
+                certToRestore = cert;
+                os = updateCert(oldAlias, cert, storeLoc, passkey, keyStore);
+                isUpdateSuccessful = true;
+            }
+
+        } catch (final IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException ex) {
+            isUpdateSuccessful = false;
+            LOG.error("Unable to update the Certifiate: ", ex.getLocalizedMessage(), ex);
+            throw new CertificateManagerException(ex.getMessage(), ex);
+        } finally {
+            closeFiles(is, os);
+        }
+        return isUpdateSuccessful;
+    }
+
+    private static FileOutputStream updateCert(String oldAlias, Certificate cert, final String storeLoc,
+        final String passkey,
+        KeyStore tstore)
+            throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+        FileOutputStream os;
+        tstore.deleteEntry(oldAlias);
+        os = new FileOutputStream(storeLoc);
+        tstore.setCertificateEntry(cert.getAlias(), cert.getX509Cert());
+        tstore.store(os, passkey.toCharArray());
+        return os;
+    }
+
+    @Override
+    public Certificate restoreCertificate() {
+        return certToRestore;
```

Comment:
Can we move the restoreCertificate in bean. It does not have any functionality .. you are just keeping a copy before updating the certificate.